### PR TITLE
Fix apache/camel-k#1119: do not return cloud events by default

### PR DESCRIPTION
**Pull request by nicolaferraro**
and deprecate automatic json serialization.

### CODE REVIEW

File: camel-knative/camel-knative/src/main/java/org/apache/camel/component/knative/KnativeReplyProcessor.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public boolean process(Exchange exchange, AsyncCallback callback) {
+        if (cloudEventEnabled) {
+            // Delegate to CloudEvent processor
+            return processor.process(exchange, callback);
+        }
+
+        // remove CloudEvent headers
+        for (CloudEvent.Attribute attr : cloudEventProcessor.cloudEvent().attributes()) {
+            exchange.getMessage().removeHeader(attr.http());
```
Comment (lburgazzoli) [inline camel-knative/camel-knative/src/main/java/org/apache/camel/component/knative/KnativeReplyProcessor.java]: shouldn't this be part of the transport processor ?  as it is the only one that know how headers are mapped

Comment (nicolaferraro) [reply camel-knative/camel-knative/src/main/java/org/apache/camel/component/knative/KnativeReplyProcessor.java]: Yeah, let me move it

**Comment by nicolaferraro** [comment]
How does it look?

**Comment by nicolaferraro** [comment]
How does it look?

**Comment by lburgazzoli** [comment]
better :)

wonder if `removeEventHeadersInReply` should be an endpoint property and a service metadata instead of a transport option

**Comment by nicolaferraro** [comment]
It seems strange as service metadata, since they are computed externally. And it tracks the `replyWithCloudEvent` endpoint option, but the `camel-knative-http` project is not aware of the `camel-knative` project where the endpoint is defined....

I also don't like it a lot.. suggestions?

**Comment by lburgazzoli** [comment]
leave it as it is for the moment, at least till we rewrite the component once more :)

**Comment by lburgazzoli** [comment]
ok to test

**Pull request merged**